\chapter{Background}\label{chap:background}
% \input{figures/background/algBackpropagation}
% \input{figures/background/figTikz}

\section{Chromosome Conformation Capture and Hi-C}\label{sec:c3}\label{sec:hic}

\input{figures/background/figsHiC}

\subsection{Overview}


Chromatin usually describes different levels of how DNA organizes itself. The
well-known double-helix is only the lowest of several structural layers.

Looking at it from the outside (highest structural layer), DNA looks similar to
a big ball of wool. With the help of Hi-C (or other methods) we can visualize
the spatial proximity.

% Gene markers tend to also have effect on their spatial neighbourhood, not only on the neighbourhood going up and down the strand of DNA.



Chromosome Conformation Technologies describe several similar methods to
compare genomic loci. They all start by:

\begin{itemize}
    \item creating chromatin cross-links (\secref{sec:crosslinking})
    \item digesting the cross-linked chromatin (\secref{sec:digestion}) and
    \item ligating the ends (\secref{sec:ligation})
\end{itemize}

to get a reversed cross-link sequence.

We will, however focus on Hi-C only, which follows up with:

\begin{itemize}
    \item shortening the crosslinks by sonication (\secref{sec:sonication}),
    \item filtering leaving only those with biotin (\secref{sec:pulldown}) and
    \item sequencing (\secref{sec:sequencing}).
\end{itemize}

The full process can be seen in \figref{fig:HiC}.

\subsection{Cross-linking DNA}\label{sec:crosslinking}

The first step is to cross-link DNA strands that are close to each other
spatially (see \figref{fig:HiC} for reference). This is done by adding
formaldehyde, which bonds sufficiently close strands together.

A chromatin cross-link is two entirely different parts of the genome held
together by a chemical bond with formaldehyde. This process cannot be
specifically controlled, so only `regions near each other' are connected, not
necessarily two `known to be close' regions.

\subsection{Digestion}\label{sec:digestion}

The next step is cutting the ball of wool apart in intervals. For this,
restriction-enzymes are used (specifically restriction endonuclease). Commonly
used enzymes for this are EcoR1 or HindIII, cutting the genome every 4000 base-pairs
\draft{info taken from wikipedia, add some other source}.

This will result in a lot of cross-linked fragments, as well as not-cross-linked ones.

\subsection{Ligation}\label{sec:ligation}

After reducing the concentration of fragments, DNA ligase is added, to ligate
(weld together) dangling fragment ends. The reduction in concentration is done
since mostly fragments close together are ligated, and we intend to ligate
fragments held together by formaldehyde. Also, Biotin is added to mark the
points of ligation. This will let us filter out a lot of fragments that have
not been ligated later on.


\subsection{Reverse Cross-links}\label{sec:revcrosslink}

% Source: http://www.protocol-online.org/biology-forums/posts/10475.html
Adding a high concentration of salt for some time will reverse the
cross-linking through formaldehyde, leaving us with our two originally
spatially close fragments ligated and with a biotin-marker.

Our fragments, however, are too long to sequence them effectively (remember
that we have now ligated fragments of around 8000 base-pairs, most sequencing
methods can only deal with sequence lengths of a few hundred base-pairs).

\subsection{Sonication}\label{sec:sonication}

The next step is to put them under influence of ultrasonic waves, breaking them
apart in much shorter fragments (due to long sequences not being able to absorb
frequent shocks well), short enough to actually enable sequencing.

\subsection{Filtering and Removal of Biotin}\label{sec:pulldown}

Here we filter all the fragments, leaving only those having a biotin-marker
(see \secref{sec:ligation}) and thus have been ligated earlier. Subsequently
we remove the marker, since it would get in the way of sequencing.

\subsection{Sequencing}\label{sec:sequencing}

Sequencing, short for DNA sequencing, describes processes of measuring a DNA
sequence. There are several techniques for doing this, most use PCR (Polymerase
Chain Reaction) before or while sequencing, which duplicates the fragments
several times, to sequence them more accurately.

% - putting the 3' and 5' ends there
% - usual PCR methods


% Open questions:
% - scaffolds?

% Chromatin is packaged into three-dimensional structures, that retain a
% relationship between genomic and physical distance. Sequences that are closer
% on the same chromosome, are also closer in physical space. Our method
% exploits this relationship between linkage and proximity to enable whole
% chromosomes scaffolding and phasing of genomes.
%
% The DNA in the sample is cross-linked in-vivo to fix DNA sequences present
% inside the same cell. Cross-linking trap sequence interactions across the
% entire genome and between different chromosomes.
%
% Cross-Linked DNA is fragmented with endonucleases. Fragmented loci are then
% biotin elated and ligated creating chimeric junctions between adjecent
% sequences. This process is called proximity ligation.
%
% The more often two sequences are joined together, the closer these two
% sequences are in genomic space.
%
% Biotinylated junctions are purified and subjected to paired-end sequencing.
% The proximity-ligation-reads are then mapped onto a draft assembly.
%
% Proximity information is used to assign context to chromosomes, and order and
% orient them along chromosome scale scaffolds.
%
% This results in fully scaffolded chromosomes of virtually any size. This
% process also detects structural variation and corrects assembly miss-joins as
% well as maps the three-dimensional conformation of chromatin within a
% population of cells.

% See \figref{fig:HiC}.


% Background:
\todo{Cite/introduce/... the given papers, and introduce the required concepts}

Required concepts:
\begin{itemize}
    \item Biology:
    \item The iterative algorithm (again ?)
    \item analysis that can be done further
    \item outlook. Meaning: What can be done, when having the 3C-Data?
\end{itemize}



\section{Further Processing}\label{sec:furtherprocessing}





